src/Apps/ArtAdvisor/02-Markdown/App.tsx

```diff
@@ -0,0 +1,152 @@
+      if (box) {
+        box.scrollTop = box.scrollHeight - box.clientHeight
+      }
+    })
+
+    await runner.finalChatCompletion()
+    setMarkdownText(current => current + "\n\n")
+    setIsGenerating(false)
+  }
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus()
+    }
+
+    // Setup: insert some CSS rules into the document's head
+    const styleElement = insertStyles()
+
+    // Cleanup: remove the style element when the component is unmounted
+    return () => removeStyles(styleElement)
+  }, [])
+
+  return (
+    <Box>
+      <Text variant={"lg-display"} my={4}>
+        Chat 1
+      </Text>
+
+      <Box
+        ref={boxRef}
+        style={{ fontSize: "1.2em" }}
+        height={"30em"}
+        border={"solid 1px"}
+        borderColor={"black60"}
+        my="2em"
+        p="1em"
+        overflow={"scroll"}
+      >
+        <Markdown>{markdownText}</Markdown>
+
+        <div ref={scrollMeRef}>
+          {/* invisible element that will be scrolled into view as `messages` gets updated */}
+        </div>
+      </Box>
+
+      <form onSubmit={handleSubmit}>
+        <Flex gap={1} mb={4}>
+          <Input
+            ref={inputRef}
+            style={{ fontSize: "1.2em" }}
+            placeholder="Chat with Artsy"
+            defaultValue="List the different printmaking techniques. For each provide the name of the technique in boldface, the definirion in italic, and a link to an example artwork for that technique on Artsy.net"
+          />
+          <Button px={6} disabled={isGenerating}>
+            Send
+          </Button>
+        </Flex>
+      </form>
+
+      <details>
+        <summary>See raw response</summary>
+
+        <textarea
+          style={{ width: "100%", height: "auto" }}
+          rows={30}
+          value={markdownText}
+          readOnly
+        />
+      </details>
+    </Box>
+  )
+}
+
+function insertStyles() {
+  // Create a new style element
+  const styleElement = document.createElement("style")
+  styleElement.type = "text/css"
+
+  // Insert the style element into the document's head
+  document.head.appendChild(styleElement)
+
+  // Add some CSS rules
+  if (styleElement.sheet) {
+    styleElement.sheet.insertRule(
+      "ul, ol { margin: 0 1em; padding: 0em; }",
+      styleElement.sheet.cssRules.length
+    )
+    styleElement.sheet.insertRule(
+      "ol li { list-style-type: number; margin-bottom: 1em; }",
+      styleElement.sheet.cssRules.length
+    )
+    styleElement.sheet.insertRule(
+      "ul li { list-style-type: disc; margin-bottom: 1em; }",
+      styleElement.sheet.cssRules.length
+    )
+    styleElement.sheet.insertRule(
+      "table { border: 1px solid gray; border-collapse: collapse; }",
+      styleElement.sheet.cssRules.length
+    )
+    styleElement.sheet.insertRule(
+      "td, th { border: 1px solid gray; padding: 0.5em; }",
+      styleElement.sheet.cssRules.length
+    )
+  }
+
+  return styleElement
+}
```

Comment:
So we don't get into any funny states later, whats this about?

Reply:
Yeah I injected some styles for our very _un_-styled lists so that the markdown would look ok.

Then I use the return value here to discard these styles when the page unmounts (via useEffect), assuming this is the good-citizen thing to do given client side nav.

I feel like there’s a better way to inject a few ad hoc style rules, which we can pursue as an improvement here.

Reply:
Could we do something like this styled-components setup? 

```tsx
const Container = styled(Box)`
  ul: ... 
`

const App = () => {
  return (
    <Container>
      ...
    </Container>
  )
}

Reply:
Good call